bigquery_storage/to_dataframe/main_test.py

```diff
@@ -0,0 +1,178 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+
+import pytest
+
+
+@pytest.fixture
+def clients():
+    # [START bigquerystorage_pandas_tutorial_create_client]
+    import google.auth
+    from google.cloud import bigquery
+    from google.cloud import bigquery_storage_v1beta1
+
+    # Explicitly create a credentials object. This allows you to use the same
+    # credentials for both the BigQuery and BigQuery Storage clients, avoiding
+    # unnecessary API calls to fetch duplicate authentication tokens.
+    credentials, project_id = google.auth.default(
+        scopes=["https://www.googleapis.com/auth/cloud-platform"]
+    )
+
+    # Make clients.
+    bqclient = bigquery.Client(credentials=credentials, project=project_id)
+    bqstorageclient = bigquery_storage_v1beta1.BigQueryStorageClient(
+        credentials=credentials
+    )
+    # [END bigquerystorage_pandas_tutorial_create_client]
+    return bqclient, bqstorageclient, project_id
+
+
+@pytest.fixture
+def temporary_dataset(clients):
+    bqclient, _, _ = clients
+    dataset_id = "bqstorage_to_dataset_{}".format(uuid.uuid4().hex)
+
+    # [START bigquerystorage_pandas_tutorial_create_dataset]
+    from google.cloud import bigquery
+
+    # TODO: Set the dataset_id to the dataset used to store temporary results.
+    # dataset_id = "query_results_dataset"
+
+    dataset_ref = bqclient.dataset(dataset_id)
+    dataset = bigquery.Dataset(dataset_ref)
+
+    # Remove tables after 24 hours.
+    dataset.default_table_expiration_ms = 1000 * 60 * 60 * 24
+
+    bqclient.create_dataset(dataset)  # API request.
+    # [END bigquerystorage_pandas_tutorial_create_dataset]
+    yield dataset_id
+    bqclient.delete_dataset(dataset_ref, delete_contents=True)
+
+
+def test_table_to_dataframe(capsys, clients):
+    bqclient, bqstorageclient, _ = clients
+
+    # [START bigquerystorage_pandas_tutorial_read_table]
+    from google.cloud import bigquery
+
+    # Download a table.
+    table = bigquery.TableReference.from_string(
+        "bigquery-public-data.utility_us.country_code_iso"
+    )
+    rows = bqclient.list_rows(
+        table,
+        selected_fields=[
+            bigquery.SchemaField("country_name", "STRING"),
+            bigquery.SchemaField("fips_code", "STRING"),
+        ],
+    )
+    dataframe = rows.to_dataframe(bqstorageclient)
+    print(dataframe.head())
+    # [END bigquerystorage_pandas_tutorial_read_table]
+
+    out, _ = capsys.readouterr()
+    assert "country_name" in out
+
+
+def test_query_to_dataframe(capsys, clients, temporary_dataset):
+    bqclient, bqstorageclient, _ = clients
+    dataset_id = temporary_dataset
+
+    # [START bigquerystorage_pandas_tutorial_read_query_results]
+    import uuid
+
+    from google.cloud import bigquery
+
+    # Due to a known issue in the BigQuery Storage API (TODO: link to
```

Comment:
Is this a todo for you?

Reply:
Yeah. I won’t submit until we have that filed publicly somewhere.

Reply:
I think it would probably be better to have it be vague (no TODO or bug link) than have a random TODO in here. I think it looks a bit unprofessional to have our TODOs show up in samples.